### PR TITLE
Multistate value init story

### PR DESCRIPTION
**Pull request by dcmaverick**
Changes to the bacnet-stack to handle multistate value init,
Jira story:
https://jira.se.com/browse/FIRMWARE-27956

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ project(
 #
 # <INSERT_CMAKE_INSTALL_PREFIX_HERE>
 #set(CMAKE_INSTALL_PREFIX "/opt/poky/rzn1/sysroots/x86_64-pokysdk-linux/usr")
+#set(CMAKE_INSTALL_PREFIX "/opt/poky/rzn1/sysroots/x86_64-pokysdk-linux/usr")
```
Comment (metelik) [inline CMakeLists.txt:14]: get a rid of this line ;) not exactlt necessary to si there....

File: src/bacnet/basic/object/msv.c [hunk 2]
```diff
@@ -328,7 +372,43 @@ bool Multistate_Value_State_Text_Set(
...
+    unsigned int j;
+
+    if((int) pInit_state_text_data->length > MSV_Max_Index) {
+        PRINT("pInit_state_text_data->length = %d >= %d", (int) pInit_state_text_data->length, MSV_Max_Index);
+        return false;
+    }
+
+    for (i = 0; i <= MSV_Max_Index; i++) {
+
+        if(i == MSV_Max_Index) {
```
Comment (metelik) [inline src/bacnet/basic/object/msv.c:397]: sometimes you insert a ' ' after if or for , sometimes you do not. Please try being consistent.

**Review by metelik — APPROVED** [review]
Couple minor things to adjust.

**Review by alanj853 — APPROVED** [review]

File: src/bacnet/basic/object/msv.c (hunk 2 above)
Comment (dcmaverick) [reply src/bacnet/basic/object/msv.c:397]: added in the space

File: CMakeLists.txt (hunk 1 above)
Comment (dcmaverick) [reply CMakeLists.txt:13]: removed

Comment (metelik) [reply CMakeLists.txt:13]: Why I do not see tidying up code here ?

File: src/bacnet/basic/object/msv.c [hunk 3]
```diff
@@ -38,6 +39,8 @@
 #include "bacnet/basic/object/msv.h"
 #include "bacnet/basic/services.h"
 
+#define PRINTF printf
```
Comment (adrigonzo) [inline src/bacnet/basic/object/msv.c:42]: if it's just a capitalisation of the same, why not just leave it as `printf` then?

File: src/bacnet/basic/object/msv.c [hunk 4]
```diff
@@ -96,21 +104,59 @@ void Multistate_Value_Init(void)
...
 /* that correlates to the correct instance number */
 unsigned Multistate_Value_Instance_To_Index(uint32_t object_instance)
 {
-    unsigned index = MAX_MULTISTATE_VALUES;
+    unsigned index = 0;
 
-    if (object_instance < MAX_MULTISTATE_VALUES) {
-        index = object_instance;
-    }
+    for (; index < MSV_Max_Index && Instance[index] != object_instance; index++) ;
```
Comment (adrigonzo) [inline src/bacnet/basic/object/msv.c:159]: I presume in this case it's ok that index returns as `MSV_Max_Index`

**Pull request merged**